### PR TITLE
Add libgif port

### PR DESCRIPTION
**Pull request by saschanaz**
This can successfully build JPEG XL reference software with GIF support.

### CODE REVIEW

File: tools/ports/libgif.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+VERSION = '5.2.1'
+HASH = '4550e53c21cb1191a4581e363fc9d0610da53f7898ca8320f0d3ef6711e76bdda2609c2df15dc94c45e28bff8de441f1227ec2da7ea827cb3c0405af4faa4736'
+
+
+def needed(settings):
+  return settings.USE_LIBGIF
+
+
+def get(ports, settings, shared):
+  ports.fetch_project('libgif', 'https://vorboss.dl.sourceforge.net/project/giflib/giflib-5.2.1.tar.gz', 'giflib-5.2.1', sha512hash=HASH)
```
Comment (kripken) [inline tools/ports/libgif.py:19]: the appearances of `5.2.1` could be replaced with e.g. `f'giflib-{VERSION}'`

**Review by kripken — COMMENTED** [review]
What is the relationship between JPEG XL and libgif? (Sorry, I don't know much about this area - when I did a search for libgif I found "giflib" which doesn't mention JPEG XL at all, is that something else?)

**Comment by saschanaz** [comment]
giflib is libgif as zlib is libz. Maybe it should be giflib here to follow zlib.

**Comment by saschanaz** [comment]
>What is the relationship between JPEG XL and libgif

https://gitlab.com/wg1/jpeg-xl depends on giflib to decode gif. I just intended to say I tested this with a real world project.

**Comment by kripken** [comment]
I see, thanks @saschanaz !

Ok, looks good, just please add a test before we land. Can be similar to the test for libjpeg for example, https://github.com/emscripten-core/emscripten/blob/63422ac7ea3d7d907c92c5d406631f060f48deb1/tests/test_other.py#L1501-L1506

**Review by kripken — APPROVED** [review]

**Pull request merged**